src/services/ActivityService.ts

```diff
@@ -49,10 +49,20 @@ export class ActivityService {
 
         // 'visit' activity validations and object field assignments
         if (activity.activityType === ActivityType.VISIT && await this.performVisitActValidations(activity)) {
-            const startTime: string = new Date().toISOString();
-            Object.assign(activity, {startTime});
-            // The endTime will be null
-            Object.assign(activity, {endTime: null});
+            if (!activity.startTime) {
```

Comment:
Could be achieved much more succinctly with ternaries

e.g. Object.assign(activity, {startTime : activity.startTime ? activity.startTime : new Date().toISOString()});
OR
const startTime = activity.startTime ? activity.startTime : new Date().toISOString()
Object.assign(activity, {startTime});

Reply:
Done.